server/src/main/java/org/elasticsearch/script/CtxMap.java

```diff
@@ -81,7 +74,8 @@ public Metadata getMetadata() {
     @Override
     public Set<Map.Entry<String, Object>> entrySet() {
         // Make a copy of the Metadata.keySet() to avoid a ConcurrentModificationException when removing a value from the iterator
-        return new EntrySet(source.entrySet(), new HashSet<>(metadata.keySet()));
+        Set<Map.Entry<String, Object>> sourceEntries = directSourceAccess() ? source.entrySet() : Set.of(new IndirectSourceEntry());
```

Comment:
So `entrySet().remove(IndirectSourceEntry instance)` will fail due to `Set.of(new IndirectSourceEntry())`.  Got it.

How does it handle `entrySet().iterator().remove()`?

Reply:
It will throw UOE because Set.of is immutable.